x/curating/keeper/post.go

```diff
@@ -195,3 +195,20 @@ func (k Keeper) CurationQueueIterator(
 		types.KeyPrefixCurationQueue,
 		sdk.PrefixEndBytes(types.CurationQueueByTimeKey(endTime)))
 }
+
+func (k Keeper) GetPosts(ctx sdk.Context) []types.Post {
+	store := ctx.KVStore(k.storeKey)
+	iterator := store.Iterator(types.KeyPrefixPost, nil)
+
+	var posts []types.Post
+
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var object types.Post
+		k.cdc.MustUnmarshalBinaryBare(iterator.Value(), &object)
```

Comment:
Having a small problem with this line.

When I query for posts:
```
stakecli query curating posts 1
```

I get the following errors.

From the client:
```
ERROR: ABCIQuery: RPC error -32603 - Internal error: proto: wrong wireType = 2 for field VendorID
```

From the node:
```
E[2020-07-08|15:14:33.320] Panic in RPC HTTP handler                    module=rpc-server err="proto: wrong wireType = 2 for field VendorID" stack="goroutine 4314 [running]:\nruntime/debug.Stack(0xc0029b10f8, 0x136a9c0, 0xc00264fca0)\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x9d\ngithub.com/tendermint/tendermint/rpc/lib/server.RecoverAndLogHandler.func1.1(0xc002e2c120, 0x196c180, 0xc000fb2ce0, 0xbfb9af325319d5cb, 0x6b61fbfba5, 0x2424b00, 0xc0030cc000)\n\t/home/meow/go/pkg/mod/github.com/tendermint/tendermint@v0.33.4/rpc/lib/server/http_server.go:161 +0x486\npanic(0x136a9c0, 0xc00264fca0)\n\t/usr/local/go/src/runtime/panic.go:969 +0x166\ngithub.com/cosmos/cosmos-sdk/codec.(*ProtoCodec).MustUnmarshalBinaryBare(0xc000e03750, 0xc00338f200, 0x16, 0x16, 0x197fd80, 0xc002ef97c0)\n\t/home/meow/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.34.4-0.20200530180557-ba70f4d4dc2e/codec/proto_codec.go:78 +0x8a\ngithub.com/cosmos/cosmos-sdk/codec.(*HybridCodec).MustUnmarshalBinaryBare(0xc000ea35c0, 0xc00338f200, 0x16, 0x16, 0x197fd80, 0xc002ef97c0)\n\t/home/meow/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.34.4-0.20200530180557-ba70f4d4dc2e/codec/hybrid_codec.go:40 +0x65\ngithub.com/public-awesome/stakebird/x/curating/keeper.Keeper.GetPosts(0x1956ea0, 0xc000e03890, 0x1988300, 0xc000e44690, 0x1970480, 0xc0004d3900, 0x7f7ba4487270, 0xc0001bf540, 0x1988300, 0xc000e44690, ...)\n\t/home/meow/Dev/RocketSpace/stakebird/x/curating/keeper/post.go:208 +0x207\ngithub.com/public-awesome/stakebird/x/curating/keeper.queryPosts(0x196b5c0, 0xc000042118, 0x1984600, 0xc000fa0e00, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, ...)\n\t/home/meow/Dev/RocketSpace/stakebird/x/curating/keeper/querier.go:65 +0x78\ngithub.com/public-awesome/stakebird/x/curating/keeper.NewQuerier.func1(0x196b5c0, 0xc000042118, 0x1984600, 0xc000fa0e00, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0, ...)\n\t/home/meow/Dev/RocketSpace/stakebird/x/curating/keeper/querier.go:28 +0x287\ngithub.com/cosmos/cosmos-sdk/baseapp.handleQueryCustom(0xc0001bf2c0, 0xc000fa0dc0, 0x4, 0x4, 0x24500a8, 0x0, 0x0, 0xc0031f2a40, 0x17, 0x5b, ...)\n\t/home/meow/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.34.4-0.20200530180557-ba70f4d4dc2e/baseapp/abci.go:475 +0x65a\ngithub.com/cosmos/cosmos-sdk/baseapp.(*BaseApp).Query(0xc0001bf2c0, 0x24500a8, 0x0, 0x0, 0xc0031f2a40, 0x17, 0x0, 0x1, 0x0, 0x0, ...)\n\t/home/meow/go/pkg/mod/github.com/cosmos/cosmos-sdk@v0.34.4-0.20200530180557-ba70f4d4dc2e/baseapp/abci.go:314 +0x386\ngithub.com/tendermint/tendermint/abci/client.(*localClient).QuerySync(0xc0002dd500, 0x24500a8, 0x0, 0x0, 0xc0031f2a40, 0x17, 0x0, 0x1, 0x0, 0x0, ...)\n\t/home/meow/go/pkg/mod/github.com/tendermint/tendermint@v0.33.4/abci/client/local_client.go:207 +0x105\ngithub.com/tendermint/tendermint/proxy.(*appConnQuery).QuerySync(0xc000e53ed0, 0x24500a8, 0x0, 0x0, 0xc0031f2a40, 0x17, 0x0, 0x1, 0x0, 0x0, ...)\n\t/home/meow/go/pkg/mod/github.com/tendermint/tendermint@v0.33.4/proxy/app_conn.go:143 +0x6a\ngithub.com/tendermint/tendermint/rpc/core.ABCIQuery(0xc002e2c160, 0xc0031f2a40, 0x17, 0x24500a8, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0, ...)\n\t/home/meow/go/pkg/mod/github.com/tendermint/tendermint@v0.33.4/rpc/core/abci.go:20 +0x114\nreflect.Value.call(0x13ae680, 0x16dbcc8, 0x13, 0x153d36d, 0x4, 0xc000fae600, 0x5, 0x5, 0x5, 0x18, ...)\n\t/usr/local/go/src/reflect/value.go:460 +0x8ab\nreflect.Value.Call(0x13ae680, 0x16dbcc8, 0x13, 0xc000fae600, 0x5, 0x5, 0x4, 0x4, 0x0)\n\t/usr/local/go/src/reflect/value.go:321 +0xb4\ngithub.com/tendermint/tendermint/rpc/lib/server.makeJSONRPCHandler.func1(0x1965000, 0xc002e2c120, 0xc0030cc000)\n\t/home/meow/go/pkg/mod/github.com/tendermint/tendermint@v0.33.4/rpc/lib/server/http_json_handler.go:103 +0x8d8\ngithub.com/tendermint/tendermint/rpc/lib/server.handleInvalidJSONRPCPaths.func1(0x1965000, 0xc002e2c120, 0xc0030cc000)\n\t/home/meow/go/pkg/mod/github.com/tendermint/tendermint@v0.33.4/rpc/lib/server/http_json_handler.go:127 +0x9c\nnet/http.HandlerFunc.ServeHTTP(0xc000fc8980, 0x1965000, 0xc002e2c120, 0xc0030cc000)\n\t/usr/local/go/src/net/http/server.go:2012 +0x44\nnet/http.(*ServeMux).ServeHTTP(0xc000217b00, 0x1965000, 0xc002e2c120, 0xc0030cc000)\n\t/usr/local/go/src/net/http/server.go:2387 +0x1a5\ngithub.com/tendermint/tendermint/rpc/lib/server.maxBytesHandler.ServeHTTP(0x1943d00, 0xc000217b00, 0xf4240, 0x1965000, 0xc002e2c120, 0xc0030cc000)\n\t/home/meow/go/pkg/mod/github.com/tendermint/tendermint@v0.33.4/rpc/lib/server/http_server.go:210 +0xcf\ngithub.com/tendermint/tendermint/rpc/lib/server.RecoverAndLogHandler.func1(0x1965880, 0xc0000f7340, 0xc0030cc000)\n\t/home/meow/go/pkg/mod/github.com/tendermint/tendermint@v0.33.4/rpc/lib/server/http_server.go:183 +0x360\nnet/http.HandlerFunc.ServeHTTP(0xc000144690, 0x1965880, 0xc0000f7340, 0xc0030cc000)\n\t/usr/local/go/src/net/http/server.go:2012 +0x44\nnet/http.serverHandler.ServeHTTP(0xc000d2c000, 0x1965880, 0xc0000f7340, 0xc0030cc000)\n\t/usr/local/go/src/net/http/server.go:2807 +0xa3\nnet/http.(*conn).serve(0xc0030783c0, 0x196b580, 0xc003098940)\n\t/usr/local/go/src/net/http/server.go:1895 +0x86c\ncreated by net/http.(*Server).Serve\n\t/usr/local/go/src/net/http/server.go:2933 +0x35c\n"
```

@shanev or @jhernandezb, any ideas of what I'm doing wrong?

Reply:
Did you reset your chain? Because `VendorID` used to be a `uint64` so it may not be the `uint32` its expecting here.

Reply:
Oh, will try that...

Reply:
Wasn't that. :sob: